bundles/CoreBundle/Command/ThumbnailsImageCommand.php

```diff
@@ -149,54 +172,88 @@ protected function execute(InputInterface $input, OutputInterface $output)
+                }
 
-        $progress = new ProgressBar($output, $totalToGenerate);
-        $progress->setFormat(
-            ' %current%/%max% [%bar%] %percent:3s%% (%elapsed:6s%/%estimated:-6s%) %memory:6s%: %message%'
-        );
-        $progress->start();
+                $items[] = serialize($item);
+            }
+        }
+        return $items;
+    }
 
-        for ($i = 0; $i < (ceil($total / $perLoop)); $i++) {
-            $list->setLimit($perLoop);
-            $list->setOffset($i * $perLoop);
-            $images = $list->load();
+    /**
+     * @inheritdoc
+     *
+     */
+    protected function runSingleCommand(string $item, InputInterface $input, OutputInterface $output)
+    {
+        $item = unserialize($item);
 
-            foreach ($images as $image) {
-                if (!$image instanceof Asset\Image) {
-                    continue;
-                }
+        $image = Asset\Image::getById($item['image_id']);
+        if (!$image instanceof Asset\Image) {
+            return;
+        }
 
-                if ($input->getOption('force')) {
-                    foreach ($thumbnailConfigNames as $thumbnailConfigName) {
-                        $image->clearThumbnail($thumbnailConfigName);
-                    }
-                }
+        $thumbnail = $item['thumbnail'] ?? null;
+        if(!$thumbnail instanceof Asset\Image\Thumbnail\Config) {
+            return;
+        }
 
-                foreach ($thumbnailsToGenerate as $thumbnailConfig) {
-                    $thumbnail = $image->getThumbnail($thumbnailConfig);
+        if (isset($item['clear_thumbnail'])) {
+            $image->clearThumbnail($thumbnail->getName());
+        }
 
-                    $progress->setMessage(
-                        sprintf(
-                            'generated thumbnail for image [%d] | file: %s',
-                            $image->getId(),
-                            $thumbnail->getPath(false)
-                        )
-                    );
+        $thumbnail = $image->getThumbnail($thumbnail);
 
-                    $progress->advance(1);
-                }
-            }
-            \Pimcore::collectGarbage();
+
+        if ($output->isVerbose()) {
+            $output->writeln(
+                sprintf(
+                    'generated thumbnail for image [%d] | file: %s',
+                    $image->getId(),
+                    $thumbnail->getPath(false)
+                )
+            );
+        } else {
+            $thumbnail->getPath(false);
         }
+    }
 
-        $progress->finish();
+    /**
+     * @inheritdoc
+     *
+     */
+    protected function runAfterBatch(InputInterface $input, OutputInterface $output)
+    {
+        \Pimcore::collectGarbage();
+    }
 
-        $output->writeln('');
+    /**
+     * @inheritdoc
+     *
+     */
+    protected function getItemName(int $count): string
+    {
+        return 1 === $count ? 'thumbnail' : 'thumbnails';
+    }
```

Comment:
Where is this used? Can't find it [in Symfony](https://github.com/symfony/symfony/search?q=ItemName&unscoped_q=ItemName) or [in Pimcore](https://github.com/pimcore/pimcore/search?q=ItemName&unscoped_q=ItemName), but it says that it is using `@inheritdoc`, inherited from where?

Reply:
please see https://github.com/webmozarts/console-parallelization/blob/8f7440d1bdb5e350a081593395cc217705401ff0/src/Parallelization.php#L203

Reply:
Oh, thanks. Why not use Symfony?
`$translate->trans('1 thumbnail|%count% thumbnails', ['%count%' => $count])`?

Reply:
Current approach would be fine for me as well, as we're not using translations anywhere in the commands anyway ... so kind of a nice to have 😉

Reply:
@brusch I think you misunderstand what the Symfony method is doing, it is not modifying language, it is modifying pluralisation. Why re-invent the wheel when we have access to a simple proper well functioning wheel?